_posts/fuss-about-eventual-consistency.md

```diff
@@ -0,0 +1,133 @@
+
+From the programming model perspective, it's mostly about complete consistency. We want all of them to be written at once without incurring partial writes. At the same time, we want all of them to be available for readers when they are all written to avoid phantom reads. For example, we don't wish cart items to be available for reads while we still write the owning shopping cart.
+
+## Distributed transactions, DCOM and CORBA
+
+But then distributed transactions came, and we got carried away. In their simplest form, distributed transactions allow a transaction between two database tables defined in different databases on different computers.
+
+We did not stop there. Different resources could be enlisted in a distributed transaction, like a database table, a message queue, and a remote service invocation. Can you imagine all those heterogeneous things in the same transaction?
+
+We even went further with protocols like DCOM or CORBA, presenting to consumers a set of distributed object instances as if they were a single instance. All of them coordinated by transactions.
+
+What could possibly go wrong?
+
+## The cloud changed everything
+
+Transactions were so engraved that for some, it was hard, if ever possible, to think about a world without transactions. But "recently," something changed. The cloud became an affordable and available option to the masses. And the cloud, primarily for the vendors' benefit, decided that (distributed) transactions were to be banned in disgrace. The underlying reason is that infrastructure in the cloud is flaky in nature, and transaction protocols were thought and designed under the assumption that [the network is reliable](https://particular.net/blog/the-network-is-reliable), the number one fallacy of distributed computing.
+
+## A new hope (cit.)
+
+The cloud also came with a neat solution to the problem: messaging, message queues, and message brokers. Instead of having so many heterogeneous resources in a single transaction, developers can use reliable messaging infrastructure to send requests to other services.
+
+Messaging allows splitting significant processes into smaller chunks and connecting them using messages. A bit of a process happens there, and then a message gets sent to a different node to continue with another bit. And so on until the process is completed. 
+
+Messaging is a way to trade consistency for availability.
+
+The proposed exchange brings eventual consistency into the mix. Because what previously was a monolithic process governed by transactions locking the requestor and forcing them to wait, it's now a set of smaller steps independent of each other with no control over consistency, making it hard for requestors to wait for completion.
```

Comment:
```suggestion
The proposed exchange brings eventual consistency into the mix. Because what previously was a monolithic process governed by transactions locking the requestor and forcing them to wait, is now a set of smaller steps independent of each other with no control over consistency, making it hard for requestors to wait for completion.
```